[docs] Remove dashboard references in beats

bmorelli25 opened the pull request:
* Removes remaining references to "dashboards" in the APM docs.
* Removes old `deprecate_dashboard_loading` attribute and replaces with `no_dashboards` (used by xpack and filebeat) where possible, and `no_dashboards_apm` where not possible.
* Cleans up a couple of the conditionals.
* Related to #1715.

Once this is approved I'll open a PR in the beats repo and make the persistent changes there.

jalvz commented on docs/copied-from-beats/command-reference.asciidoc:
```diff
@@ -15,30 +15,32 @@
 
 :deploy-command-short-desc: Deploys the specified function to your serverless environment
+
+ifndef::no_dashboards_apm[]
```
L19-L21 should be removed?
Looks a dup of L23-25, same thing happens on several places bellow

bmorelli25 replied:
L19-L21 is an `ifndef` (if not defined) - this line shows up *outside* of the APM docs
L23-25 is an `ifdef` (if defined) - this line shows up *in* the APM docs

jalvz commented on docs/copied-from-beats/command-reference.asciidoc:
```diff
@@ -15,30 +15,32 @@
+endif::no_dashboards_apm[]
 
+ifndef::has_ml_jobs,no_dashboards_apm[]
+:setup-command-short-desc: Sets up the initial environment, including the index template and {kib} dashboards (when available)
```
the `{kib} dashboards (when available)` part should be removed? (sorry if not, not sure i am getting this right)

bmorelli25 replied:
All good :)
L43 only shows up if `has_ml_jobs` *and* `no_dashboards_apm` are not defined. Since we have defined `no_dashboards_apm`, L39 shows up in our documentation

bmorelli25 commented on docs/copied-from-beats/command-reference.asciidoc:
```diff
@@ -15,30 +15,32 @@
-ifdef::deprecate_dashboard_loading[]
-
-:setup-command-short-desc: Sets up the initial environment, including the ES index template and {kib} dashboards (deprecated).
+ifdef::no_dashboards_apm[]
```
Ideally we'll be able to use only `no_dashboards` for all instances. The problem is that `no_dashboards` is also used by xpack and filebeat. I need to confirm that this change is OK for them. If it is, this attribute will be changed to `no_dashboards`. If it isn't, it will stay APM sepecific.

bmorelli25 replied:
Good news, we're running with `no_dashboards` for everything!

bmorelli25 commented on docs/copied-from-beats/command-reference.asciidoc:
```diff
@@ -524,20 +549,18 @@ the end of the file is reached. By default harvesters are closed after
-ifndef::has_ml_jobs[]
+endif::has_ml_jobs[]
+ifdef::no_dashboards[]
+Loads the initial setup, including Elasticsearch template and {kib} index pattern.
```
Ahh, good find. Fixed - thank you!

bmorelli25 commented:
FYI: Still waiting on final review in the beats repo to push this change. Will hopefully combine these changes with the `observer.agent` change and merge at the same time!

bmorelli25 commented:
Rebased and updated the `no_dashboards_apm` attribute to be `no_dashboards`. Everything looks good! Still waiting on beats repo changes, but going to merge this so we can start to see the changes in the docs.
